src/preciceMap.cpp

```diff
@@ -53,13 +54,43 @@ std::vector<std::string> getMeshes(std::string basename, int rank)
   return meshes;
 }
 
-struct Mesh {
+class Mesh {
+public:
   std::vector<std::array<double, 3>> positions;
   std::vector<std::array<size_t, 2>> edges;
   std::vector<std::array<size_t, 3>> triangles;
   std::vector<double> data;
 };
 
+class VectorMesh : public Mesh{
+public:
+  VectorData data; //Overload data with Vector data
```

Comment:
An alternative implementation could be to simply use `std::vector<double> data` with a `size_t dataStride = 1` in `Mesh`. This way you just need to call `writeBlockScalarData` or `writeBlockVectorData` later.

This would save you to go through the trouble of adding an extra class and reduce code duplication.
Remember that this tool should be as minimal and lightweight as possible. (Someone needs to maintain it.)